lib/plugins/renderer/nunjucks.js

```diff
@@ -0,0 +1,62 @@
+
+'use strict';
+
+const nunjucks = require('nunjucks');
+const fs = require('hexo-fs');
+const { dirname } = require('path');
+
+function toArray(value) {
+  if (Array.isArray(value)) {
+    // Return if given value is an Array
+    return value;
+  } else if (typeof value.toArray === 'function') {
+    return value.toArray();
+  } else if (value instanceof Map) {
+    const arr = [];
+    value.forEach(v => arr.push(v));
+    return arr;
+  } else if (value instanceof Set || typeof value === 'string') {
+    return [...value];
+  } else if (typeof value === 'object' && value instanceof Object && Boolean(value)) {
+    return Object.values(value);
+  }
+
+  return [];
+}
+
+function safeJsonStringify(json, spacer = undefined) {
+  if (typeof json !== 'undefined' && json !== null) {
+    return JSON.stringify(json, null, spacer);
+  }
+
+  return '""';
+}
+
+function njkCompile(data) {
+  const env = nunjucks.configure(dirname(data.path), {
+    autoescape: false,
+    throwOnUndefined: false,
+    trimBlocks: false,
+    lstripBlocks: false
+  });
+  env.addFilter('toArray', toArray);
+  env.addFilter('safeDump', safeJsonStringify);
```

Comment:
```suggestion
  env.addFilter('toarray', toArray);
  env.addFilter('safedump', safeJsonStringify);
```
I suggest to use lowercase instead of camel case naming, which is consistent with Nunjucks built-in filter naming style
https://mozilla.github.io/nunjucks/templating.html#builtin-filters

Reply:
@stevenjoezhang But they also have `asyncAll` & `asyncEach` through.

Anyway, I will change to lowercase then.